bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1627,21 +1628,23 @@ OperationResult PreProcessor::launchReqPreProcessing(const PreProcessRequestMsgS
     LOG_ERROR(logger(),
               "Pre-execution failed" << KVLOG(
                   cid, clientId, reqOffsetInBatch, reqSeqNum, (uint32_t)preProcessResult, resultLen));
-    return preProcessResult;
   }
-  if (request.outActualReplySize == 0) {
+
+  if (preProcessResult == OperationResult::SUCCESS && request.outActualReplySize == 0) {
```

Comment:
Please leave the previous condition as is:
if (request.outActualReplySize == 0)